lib/horoshop.rb

```diff
@@ -10,7 +14,35 @@ def initialize(url:, username:, password:)
     @expiration_timestamp = nil
   end
 
+  def setup_connection
```

Comment:
I cant see how work re-authorization if token expired. And in my opinion setup connection is different method and should be separated into Horoshop::Connection.status  that return true or false